### DISCUSSION TITLE
Проблема импорта имен исполнителей на примере Lastfm.getSimilarArtists

### DISCUSSION THREAD

**Discussion by amd64fox** [Chimildic/goofy#64] [Помощь с ошибкой] [answered]
Для примера, хочу получить похожих исполнителей по [Warthog](https://www.last.fm/ru/music/Warthog), что бы смоделировать мою ошибку можно сразу запустить код ниже.

```javascript
function Url_track(){

let track = SpotifyRequest.getFullObjByIds('tracks', [Helper.parseId('spotify:track:4ffOMk4bWgzSlsX5GuSUGZ')], 1)[0];

let similarArtists = Lastfm.getSimilarArtists([track], 0.10, 20);

let TracksTop = Source.getArtistsTopTracks(similarArtists, false);
  TracksTop.forEach(tracks => tracks.splice(1, 10));
  TracksTop = TracksTop.flat(1);
  Filter.dedupArtists(TracksTop);

   Playlist.saveAsNew({
      name: 'Похожие исполнители',
      tracks: TracksTop,
      });
```
После выполнения скрипта в выдаче сразу первыми должны идти исполнители у которых противоположные жанры от нашего артиста, если обратить внимание на [первого](https://open.spotify.com/track/245onUPHGD1DYuiacxTuW0?si=BmMwe6ELSAKtgtkBdu8Pdw), то имя очень похоже на то которое сам рекомендует на странице [Last.fm](https://www.last.fm/ru/music/Warthog) только там оно `Bib` а этот `BIBI`. 

Просто хотелось понять, это уже при поиске в Spotify такая выдача кривая или где-то еще ошибка ?

**Comment by Chimildic** [comment]
Goofy берет первый результат поиска исходя из гипотезы, что он наилучший, наиболее подходящий.
Поиск Spotify по `bib` первым дает исполнителя `BIBI`, вторым `Bib`.

**Reply by amd64fox** [reply]
Блин так и думал, я так понимаю этот поиск который идет через api spotify как-то отличается от того же поиска в web-версии клиента, т.к.  если там вбить `bib`, то этот исполнитель `BIBI` даже не на 2-ом месте.

![Без3333 имени](https://user-images.githubusercontent.com/62529699/111029919-70b22b80-8410-11eb-8112-a3c7324d6450.png)

**Reply by Chimildic** [reply]
Одному Spotify известно как это работает

**Comment by Chimildic** [comment, marked as answer]
В обновленной версии 1.4.7 попытался чуть исправить ситуацию: пройдет цикл по топ 20 результатам от Spotify и выберет то, что совпадает добуквенно и по размеру. Если такого нет, вернет первый из результатов. 

В тесте в этом случае приходит `bib`, а не `bibi`.
```js
let artist = Search.multisearchArtists(['bib'], (item) => item);
```

**Reply by amd64fox** [reply]
Спасибище))  буду тестировать ...
